src/http/request.cr

```diff
@@ -98,6 +98,19 @@ class HTTP::Request
     value
   end
 
+  # Return request host from headers
+  def host
+    host = @headers["Host"]?
+    return unless host
+    index = host.index(":")
+    index ? host[0...index] : host
```

Comment:
Don't know if this is anything useful, but I played around a bit with [different implementations and benchmarking](https://gist.github.com/paddor/2f1adbed7e42ea18a1b1ba7e1bfbe690).